File: Python/Arrays/pg0002.py
#Reverse array – Rearrange elements so the first becomes last and vice versa, without extra space.

n = int(input())
arr = list(map(int,input().split()))

"""
print(arr[::-1])
"""

left = 0
right = n-1
while(left<right):
    arr[left],arr[right] = arr[right],arr[left]
    left+=1
    right-=1
print(arr)

File: Java/Arrays/pg0002.java
//Reverse array – Rearrange elements so the first becomes last and vice versa, without extra space.

import java.util.*;

class pg0002{
    public static void main(String[] Args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        if(n==0){
            System.out.println("Array is empty");
            return;
        }
        int arr[] = new int[n];
        for(int i=0;i<n;i++){
            arr[i] = sc.nextInt();
        }
        int start = 0, end = n-1;
        while(start<end){
            int temp = arr[start];
            arr[start] = arr[end];
            arr[end] = temp;
            start++;
            end--;
        }
        for(int i:arr){
            System.out.print(i+" ");
        }
    }
}
